### PR TITLE
Tsan related changes

### PR DESCRIPTION
**Pull request by icanzilb**
@freak4pc - I enabled TSan for all projects, for miniscule projects like our examples there is no overhead when compiling/running but it'll help us catch any issues early for future editions of the book.

### CODE REVIEW

**Comment by freak4pc** [comment]
Hey, did you make sure it's all working correctly? 
Seems like the integration there is still quite flaky: https://twitter.com/dgregor79/status/1482419943933034496

File: 10-advanced-actor/projects/final/Sky/Utility/Utility.swift [hunk 1]
```diff
@@ -48,7 +48,7 @@ extension Task where Success == Never, Failure == Never {
   /// Suspends the current task for at least the given duration in seconds.
   /// - Parameter seconds: The sleep duration in seconds.
   static func sleep(seconds: TimeInterval) async {
-    await Task.sleep(UInt64(seconds * 1_000_000_000))
+		try? await Task.sleep(nanoseconds: UInt64(seconds * 1_000_000_000))
```
Comment (freak4pc) [inline 10-advanced-actor/projects/final/Sky/Utility/Utility.swift]: indentation (below also)

File: 07-concurrent-code-with-taskgroup/projects/final/Sky/Utility/Utility.swift [hunk 2]
```diff
@@ -59,13 +59,15 @@ actor UnreliableAPI {
...
 
-  static func action(failingEvery: Int) throws {
+  var counter = 0
+
+  func action(failingEvery: Int) throws {
     counter += 1
     if counter % failingEvery == 0 {
       counter = 0
-      throw Error()
+			throw Error()
```
Comment (freak4pc) [inline 07-concurrent-code-with-taskgroup/projects/final/Sky/Utility/Utility.swift]: indentation

**Comment by icanzilb** [comment]
> Hey, did you make sure it's all working correctly?
> Seems like the integration there is still quite flaky: https://twitter.com/dgregor79/status/1482419943933034496

@freak4pc yeah - I built and ran all the projects, looked good to me.

If you review the code changes, there are very few of them and they're simple. Most of the PR is enabling tsan in Xcode's scheme to guard against future regressions

**Comment by icanzilb** [comment]
btw, that's why I also tagged @jrturton - would be great if he'd skim the code over as well

**Comment by jrturton** [comment]
Crikey Marin, give me a minute 😅 I'm at work!

**Comment by icanzilb** [comment]
> Crikey Marin, give me a minute 😅 I'm at work!

What is this work you're speaking of? Sounds boring

**Comment by jrturton** [comment]
Looks good to me once you fix the indentation issues Shai pointed out

**Comment by icanzilb** [comment]
Thanks @jrturton — @freak4pc https://github.com/raywenderlich/mcon-materials/pull/18/commits/0121780c0e875b79bfbd18c442e4bc5597b6806e ought to fix those two indent issues

**Review by freak4pc — APPROVED** [review]

**Pull request merged**